puma/utils/discriminant.py

```diff
@@ -0,0 +1,153 @@
+"""Discriminant calculation for flavour tagging."""
+import numpy as np
+
+from puma.utils.histogram import save_divide
+
+
+def calc_disc(
+    scores: np.ndarray,
+    flvs: list = None,
+    flv_map: dict = None,
+) -> np.ndarray:
+    """Calculate arbitrary flavour tagging score.
+
+    Parameters
+    ----------
+    scores : np.ndarray
+        tagger scores in the shape (n_jets, n_flavours)
+    flvs : list, optional
+        , by default None
+    flv_map : dict, optional
+        flavour map containing signal and background mapping from `flv` and their
+        fractions, by default None
+
+    Returns
+    -------
+    np.ndarray
+        discriminant values
+
+    Raises
+    ------
+    ValueError
+        if scores and shapes have different shapes
+
+    Examples
+    --------
+    The `flv_map` can e.g. look like this together with the `flvs`
+    >>> flv_map ={
+    ...     "sig": {"b": 1.0},
+    ...     "bkg": {"l": 1 - 0.5, "c": 0.5},
+    ... }
+    >>> flvs = ["l", "c", "b"]
```

Comment:
Here, `flvs` represents the flavours, while before it was the flavour probabilites. Maybe better to used different variable / attribute names

Reply:
I changed the flavours before as you mentioned